tools/Dashboard/DevHome.Dashboard/ViewModels/WidgetViewModel.cs

```diff
@@ -81,22 +83,13 @@ public partial class WidgetViewModel : ObservableObject
         }
     }
 
-    partial void OnWidgetFrameworkElementChanged(FrameworkElement value)
-    {
-        if (WidgetFrameworkElement != null && WidgetFrameworkElement is Grid grid)
-        {
-            WidgetBackground = grid.Background;
-        }
-    }
-
     public WidgetViewModel(

```

Comment:
Slightly off topic but I'm not sure how/when you'll do it but as you continue the efforts to refactor/ makes things more MVVM-ish in the dashboard, do you have plans on making this view model a service? I'm just looking at these parameters below and I'm wondering if it's possible to consolidate the widget items into a wrapper that can be injected. Not sure how we'd accomplish this with the dispatcher though, but I guess the question there is do we need to pass it as an argument at all?

All else fails we could have a generic dispatcher service with a dispatcher property for a view. When a view is constructed update that property in the service with the view's dispatcher queue. Then we can use the dispatcher service throughout the dashboard to get the dispatcher just by doing something like this.

```C#
var dispatcherService = Application.Current.GetService<IViewDispatcherService>();
var dashBoardDispatcher = dispatcherService.GetDispatcher(nameof(DashboardView))

// now try to enqueue your stuff
```

you could probably inject it a different way, or maybe not even do this at all, just a thought I had is all.

Reply:
This is something I have yet to really dig into. I've thought it might be a good idea to have a widget model separate from the widget view model (which sounds kind of like your "consolidate widget items into a wrapper" idea). I think @AmelBawa-msft thrown around the idea of a dispatcher service. But I do think this area needs work.